packages/sign-client/src/controllers/engine.ts

```diff
@@ -134,6 +134,7 @@ export class Engine extends IEngine {
           session.self.publicKey = publicKey;
           const completeSession = {
             ...session,
+            pairingTopic,
```

Comment:
The variable assignment flow here is unnecessarily confusing, but `pairingTopic` is only an optional parameter for `connect`, which is assigned to `let topic` above.

If `pairingTopic` isn't provided (i.e. we're connecting without a previously known pairing) we create a new one and assign that new topic to `topic` here: https://github.com/WalletConnect/walletconnect-monorepo/pull/2082/files#diff-04e05e19b9270d1267b68d324e16997e4465d6879324fe4427a6194c00410eabR108

Hence `topic` is the source of truth on the final pairing topic here:

https://github.com/WalletConnect/walletconnect-monorepo/pull/2082/files#diff-04e05e19b9270d1267b68d324e16997e4465d6879324fe4427a6194c00410eabR108

```suggestion
            pairingTopic: topic,
```